src/graph/RadarTrackObjectsNode.cpp

```diff
@@ -88,41 +100,179 @@ void RadarTrackObjectsNode::enqueueExecImpl()
+
+	const auto radarVelocity = input->getLinearVelocity();
+	const auto relVelocity = absVelocity - Vec2f{radarVelocity.x(), radarVelocity.y()};
+
+	if (objectStatus == ObjectStatus::Measured) {
+		assert(currentTimeMs <= std::numeric_limits<decltype(objectState.creationTime)>::max());
+		objectState.lastMeasuredTime = static_cast<decltype(objectState.creationTime)>(currentTimeMs);
+	}
+	objectState.objectStatus = objectStatus;
+	objectState.movementStatus = displacement.length() > movementSensitivity ? MovementStatus::Moved :
+	                                                                           MovementStatus::Stationary;
+	objectState.position.addSample(updatedPosition);
+
+	// There has to be at leas one abs velocity sample from previous frames - in other words, this has to be the third frame to be
+	// able to calculate acceleration (first frame - position, second frame - velocity, third frame - acceleration).
+	if (objectState.absVelocity.getSamplesCount() > 0) {
+		const auto absAccel = (absVelocity - objectState.absVelocity.getLastSample()) * deltaTimeSecInv;
+		objectState.absAccel.addSample(absAccel);
+
+		const auto relAccel = (relVelocity - objectState.relVelocity.getLastSample()) * deltaTimeSecInv;
+		objectState.relAccel.addSample(relAccel);
+	}
+	objectState.absVelocity.addSample(absVelocity);
+	objectState.relVelocity.addSample(relVelocity);
+
+	// Behaves similar to acceleration. In order to calculate orientation I need velocity, which can be calculated starting from
+	// the second frame. For this reason, the third frame is the first one when I am able to calculate orientation rate. Additionally,
+	// if object does not move, keep its previous orientation.
+	const auto orientation = objectState.movementStatus == MovementStatus::Moved ? atan2(absVelocity.y(), absVelocity.x()) :
+	                                                                               objectState.orientation.getLastSample();
+	if (objectState.orientation.getSamplesCount() > 0) {
+		objectState.orientationRate.addSample(orientation - objectState.orientation.getLastSample());
+	}
+	objectState.orientation.addSample(orientation);
+
+	if (objectStatus == ObjectStatus::Measured) {
+		objectState.length.addSample(updatedAabb.maxCorner().x() - updatedAabb.minCorner().x());
+		objectState.width.addSample(updatedAabb.maxCorner().y() - updatedAabb.minCorner().y());
+	} else {
+		objectState.length.addSample(objectState.length.getLastSample());
+		objectState.width.addSample(objectState.width.getLastSample());
+	}
+}
+
+void RadarTrackObjectsNode::UpdateOutputData()
+{
+	fieldData[XYZ_VEC3_F32]->resize(objectStates.size(), true, false);
+	auto* xyzPtr = static_cast<Vec3f*>(fieldData[XYZ_VEC3_F32]->getRawWritePtr());
+
+	fieldData[ENTITY_ID_I32]->resize(objectStates.size(), true, false);
+	auto* idPtr = static_cast<int32_t*>(fieldData[ENTITY_ID_I32]->getRawWritePtr());
```

Comment:
```suggestion
	fieldData[XYZ_VEC3_F32]->resize(objectStates.size(), false, false);
	auto* xyzPtr = static_cast<Field<XYZ_VEC3_F32>::type*>(fieldData[XYZ_VEC3_F32]->getRawWritePtr());

	fieldData[ENTITY_ID_I32]->resize(objectStates.size(), false, false);
	auto* idPtr = static_cast<Field<ENTITY_ID_I32>::type*>(fieldData[ENTITY_ID_I32]->getRawWritePtr());
```
1. I see no reason to zero the array
2. It should be safer to get the type of field instead of hardcoding it